### PR TITLE
Phase mode interfaces.

### PR DESCRIPTION
**Pull request by boguslaw-wojcik**

### CODE REVIEW

**Review by Brzezoo — CHANGES_REQUESTED** [review]

File: device_services/generic/chargepoint.md [hunk 1]
```diff
@@ -51,6 +55,7 @@ Chargepoint service is used to represent EV chargers.
 | `sup_states`         | str_array | `["disconnected", "charging", "ready_to_charge", "finished"]` | List of possible states of the chargepoint. See [the list](#definitions) of possible values. |
 | `sup_charging_modes` | str_array | `["slow", "normal"]`                                          | Optionally supported charging modes.                                                         |
 | `sup_max_current`    | int       | `32`                                                          | Maximum current limit in `A` as set by the installer.                                        |
+| `sup_phase_modes`    | str_array | `["NL1L2L3", "NL1"]`                                          | Optionally supported phase modes. See [the list](#definitions) of possible values.           |
```
[the remark this answers is not in the document]
Comment (boguslaw-wojcik) [reply device_services/generic/chargepoint.md:58]: You're wrong. It must indicate difference between IT L1+L2+L3 mode and TT/TN mode N+L1+L2+L3.

**Review by Brzezoo — APPROVED** [review]

File: device_services/generic/chargepoint.md [hunk 2]
```diff
@@ -51,6 +55,7 @@ Chargepoint service is used to represent EV chargers.
...
 | `grid_type`          | string    | `"TN"`                                                        | Grid type of the charger. Possible values are `IT`, `TT` and `TN`.                           |
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:59]: What about other grid types ? Not supported yet ? Shall we have "not supported" or "other" status ?

File: device_services/generic/chargepoint.md [hunk 3]
```diff
@@ -60,6 +65,22 @@ Chargepoint service is used to represent EV chargers.
 
 > Please note that `ready_to_charge` and `charging` are states required for charging control, while others have only informative value.
 
+* `phase_mode` defines allowed phase balancing modes, see table below for more details:
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:68]: We shall add some descirpion, what is Input EVSE, Output EV(SE?)...

File: device_services/generic/chargepoint.md [hunk 4]
```diff
@@ -60,6 +65,22 @@ Chargepoint service is used to represent EV chargers.
...
+
+| Value     | Input EVSE | Output EV  | Grid       |
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:70]: Grid here means, that it is only supported in the listed grid types ? Shall be explained...

Comment (BudzikM) [inline device_services/generic/chargepoint.md:70]:
```suggestion
| Value     | Input EVSE | Output EV  | Supported grids       |
```

File: device_services/generic/chargepoint.md [hunk 5]
```diff
@@ -60,6 +65,22 @@ Chargepoint service is used to represent EV chargers.
...
+| `NL1L2L3` | N L1 L2 L3 | N L1 L2 L3 | `TN`, `TT` |
+| `NL1`     | N L1       | N L1       | `TN`, `TT` |
+| `NL2`     | N L2       | N L1       | `TN`, `TT` |
+| `NL3`     | N L3       | N L1       | `TN`, `TT` |
+| `L1L2L3`  | L1 L2 L3   | N L1 L2    | `IT`       |
+| `L1L2`    | L1 L2      | N L1       | `IT`       |
+| `L2L3`    | L2 L3      | N L1       | `IT`       |
+| `L3L1`    | L1 L3      | N L1       | `IT`       |
+
+> Based on device capabilities and its configuration a chargepoint may support only a subset of the above modes.
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:81]:
```suggestion
> Based on a EVSE capabilities and its configuration a chargepoint may support only a subset of the above modes. 
```

File: device_services/generic/chargepoint.md [hunk 6]
```diff
@@ -60,6 +65,22 @@ Chargepoint service is used to represent EV chargers.
...
+| `NL1`     | N L1       | N L1       | `TN`, `TT` |
+| `NL2`     | N L2       | N L1       | `TN`, `TT` |
+| `NL3`     | N L3       | N L1       | `TN`, `TT` |
+| `L1L2L3`  | L1 L2 L3   | N L1 L2    | `IT`       |
+| `L1L2`    | L1 L2      | N L1       | `IT`       |
+| `L2L3`    | L2 L3      | N L1       | `IT`       |
+| `L3L1`    | L1 L3      | N L1       | `IT`       |
+
+> Based on device capabilities and its configuration a chargepoint may support only a subset of the above modes. 
+> For example a chargepoint in TN installation capable to switch between 3 phases and 1 phase only will support `NL1L2L3` and `NL1` modes.
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:82]:
```suggestion
> For example a chargepoint in TN grid type capable to switch between either 3 phases or 1 phase will support `NL1L2L3` and `NL1` modes.
```

File: device_services/generic/chargepoint.md (hunk 4 above)
Comment (BudzikM) [reply device_services/generic/chargepoint.md:70]: OR 'Grid' -> 'Allowed in grid types' if Supported grid types is not self-explanatory

File: device_services/generic/chargepoint.md [hunk 7]
```diff
@@ -27,6 +27,10 @@ Chargepoint service is used to represent EV chargers.
 | in   | cmd.max_current.set             | int        |                                                                     | Sets the maximum offered current in `A`, must be an integer between `6` and `sup_max_current` service property value. |
 | in   | cmd.max_current.get_report      | null       |                                                                     | Requests the maximum offered energy.                                                                                  |
 | out  | evt.max_current.report          | int        |                                                                     | Reports the maximum offered current in `A`, this is effectively a static load balancing value.                        |
+| -    |                                 |            |                                                                     |                                                                                                                       |
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:30]: 'max_current' is referred but never define in this service

File: device_services/generic/chargepoint.md (hunk 2 above)
Comment (boguslaw-wojcik) [reply device_services/generic/chargepoint.md:59]: Irrelevant for this PR. But IT, TN and TT are all grids types out there. TNS, TNC and TNCS all fall under TN type.

File: device_services/generic/chargepoint.md (hunk 7 above)
Comment (boguslaw-wojcik) [reply device_services/generic/chargepoint.md:30]: What? It's not part of this PR.

File: device_services/generic/chargepoint.md (hunk 6 above)
Comment (boguslaw-wojcik) [reply device_services/generic/chargepoint.md]: See new version. Your rewording is not consistent with the original meaning.

File: device_services/generic/chargepoint.md (hunk 4 above)
Comment (boguslaw-wojcik) [reply device_services/generic/chargepoint.md]: Maybe now it's easier to understand.

**Review by BudzikM — APPROVED** [review]

File: device_services/generic/chargepoint.md [hunk 8]
```diff
@@ -60,6 +65,23 @@ Chargepoint service is used to represent EV chargers.
 
 > Please note that `ready_to_charge` and `charging` are states required for charging control, while others have only informative value.
 
+* `phase_mode` defines allowed phase balancing modes for chargepoints installed in 3-phase configuration, see table below for more details on modes:
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md:68]: chargpoints -> EVSE ?

**Review by BudzikM — APPROVED** [review]

File: device_services/generic/chargepoint.md [hunk 9]
```diff
@@ -8,25 +8,29 @@ Chargepoint service is used to represent EV chargers.
...
+| in   | cmd.max_current.get_report      | null       |                                                                     | Requests the maximum offered energy.                                                                                      |
+| out  | evt.max_current.report          | int        |                                                                     | Reports the maximum offered current in `A`, this is effectively a static load balancing value.                            |
+| -    |                                 |            |                                                                     |                                                                                                                           |
+| in   | cmd.phase_mode.set              | string     |                                                                     | Sets the phase mode for chargepoints installed in 3-phase configuration. See [the list](#definitions) of possible values. |
```
Comment (BudzikM) [inline device_services/generic/chargepoint.md]:
```suggestion
| in   | cmd.phase_mode.set              | string     |                                                                     | Sets the phase mode for EVSE installed in 3-phase configuration. See [the list](#definitions) of possible values. |
```

**Review by BudzikM — APPROVED** [review]

**Pull request merged**